base/errorshow.jl

```diff
@@ -899,19 +899,26 @@ function show(io::IO, ip::InterpreterIP)
     end
 end
 
-# handler for displaying a hint in case the user tries to call
-# the instance of a number (probably missing the operator)
-# eg: (1 + 2)(3 + 4)
-function noncallable_number_hint_handler(io, ex, arg_types, kwargs)
+function hint_handler(io, ex, arg_types, kwargs)
     @nospecialize
+    # Display a hint in case the user tries to call the instance of a number
+    # (probably missing the operator)
+    # eg: (1 + 2)(3 + 4)
     if ex.f isa Number
         print(io, "\nMaybe you forgot to use an operator such as ")
         printstyled(io, "*, ^, %, / etc. ", color=:cyan)
         print(io, "?")
     end
+    # Display a hint in case the user tries to use the + operator on strings
+    # (probably attempting concatenation)
+    if (ex.f == +) && all(i -> i <: AbstractString, arg_types)
```

Comment:
Maybe put this in a separate function with it's own call to `Experimental.register_error_hint`?

Reply:
Done!